### PR TITLE
fix: release obj from LifecycleUtil after obj destroy

### PR DESCRIPTION
**Pull request by killagu**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message follows commit guidelines

### CODE REVIEW

**Comment by killagu** [comment]
- @eggjs/aop-decorator@3.2.4
 - @eggjs/tegg-aop-runtime@3.2.4
 - @eggjs/tegg-background-task@3.2.4
 - @eggjs/controller-decorator@3.2.4
 - @eggjs/tegg-dynamic-inject-runtime@3.2.4
 - @eggjs/tegg-eventbus-runtime@3.2.4
 - @eggjs/tegg-lifecycle@3.2.4
 - @eggjs/tegg-loader@3.2.4
 - @eggjs/tegg-metadata@3.2.4
 - @eggjs/tegg-orm-decorator@3.2.4
 - @eggjs/tegg-runtime@3.2.4
 - @eggjs/tegg-schedule-decorator@3.2.4
 - @eggjs/tegg@3.2.4
 - @eggjs/tegg-aop-plugin@3.2.4
 - @eggjs/tegg-controller-plugin@3.2.4
 - @eggjs/tegg-eventbus-plugin@3.2.4
 - @eggjs/tegg-orm-plugin@3.2.4
 - @eggjs/tegg-schedule-plugin@3.2.4
 - @eggjs/tegg-plugin@3.2.4
 - @eggjs/tegg-standalone@3.2.4

File: core/metadata/src/factory/LoadUnitFactory.ts [hunk 1]
```diff
@@ -48,6 +48,7 @@ export class LoadUnitFactory {
     } finally {
       this.loadUnitMap.delete(loadUnit.unitPath);
       this.loadUnitIdMap.delete(loadUnit.id);
+      LoadUnitLifecycleUtil.clearObjectLifecycle(loadUnit);
```
Comment (fengmk2) [inline core/metadata/src/factory/LoadUnitFactory.ts:51]: 手动清理的地方有点多。

Comment (killagu) [reply core/metadata/src/factory/LoadUnitFactory.ts:51]: 得搞个通用的工厂。这个改造也挺大了。

Comment (killagu) [reply core/metadata/src/factory/LoadUnitFactory.ts:51]: 这个是 ioc 本来就带的功能，现在这里做的是要解决管理 ioc 的对象的自动释放。

**Pull request merged**